src/Dialog/Dialog.js

```diff
@@ -295,10 +295,19 @@ class DialogInline extends React.Component {
       </div>
     );
 
-    const titleElement = typeof title === 'string' ?
-      <h3 className={titleClassName} style={prepareStyles(styles.title)}>
-        {title}
-      </h3> : title;
+    let titleElement = title;
+    if (React.isValidElement(title)) {
+      titleElement = React.cloneElement(title, {
+        className: title.props.className || titleClassName,
+        style: prepareStyles({...styles.title, ...title.props.style}),
```

Comment:
We are using the `Object.assign` instead of the destructuring on the rest of the code base. Could you do the same here?